### PR TITLE
Dockerizing the build process to make it simpler, README clean-up

### PR DESCRIPTION
**Pull request by lguminski**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,67 +11,13 @@ launching the framework <i>scheduler</i>.
...
-
-You can also build sub-components by remembering to use a relative path to the gradle wrapper. If you don't want to deal with relative paths you can run <a href="https://github.com/dougborg/gdub">gdub</a> which runs the gradle wrapper from any subdirectory.
-
-````
-$ cd /vagrant
-$ cd commons
-$ sudo ../gradlew build
-````
+```
+$ docker run --rm -v /var/run/docker.sock:/var/run/docker.sock \
```
Comment (mwl) [inline README.md]: I'm afraid we'll have to add instruction on how to build on Mac with Boot2docker/Docker machine as well.

Comment (lguminski) [inline README.md]: I don't think so. Docker is simply a prerequisite, and users need to follow instructions how to install Docker at the given platform. Explaining this again is not in scope of this project.

Comment (mwl) [inline README.md]: I agree that Docker is a prerequisite, but we can't assume that `docker.sock` is available on all platforms. So we need instructions on how to build through Docker http api.

Comment (lguminski) [inline README.md]: >  we can't assume that docker.sock is available on all platforms

Could you give an example?

Comment (mwl) [inline README.md]: OS X with Docker-machine doesn't have `docker.sock`.

Comment (lguminski) [inline README.md]: In fact it does. Because docker runs in the context of boot2docker (virtualbox) image, and it is there (inside the VM).

**Pull request closed without merging**